pinot-query-planner/src/main/java/org/apache/pinot/query/routing/table/LogicalTableRouteInfo.java

```diff
@@ -234,15 +238,15 @@ public boolean hasRealtime() {
   @Nullable
   @Override
   public String getOfflineTableName() {
-    return hasOffline() && _logicalTable != null ? TableNameBuilder.OFFLINE.tableNameWithType(
-        _logicalTable.getTableName()) : null;
+    return hasOffline() && _logicalTableName != null ? TableNameBuilder.OFFLINE.tableNameWithType(_logicalTableName)
```

Comment:
How can `_logicalTableName` be null?

should it be set during the init or constructor? Its too basic.

Reply:
This can happen if a table name that does not exist is provided. I have kept the same semantics as `ImplicitHybridTableRouteInfo` where the table config is checked for null to see if it exists. The route provider returns a TableRouteInfo always instead of a null or an exception. A `TableRouteInfo` is returned to keep the logic same in `BaseSingleStageRequestHandler`